### PR TITLE
fix: added blockage of unnecessary event passthrough on datarow

### LINKED ISSUE #396: DataRow's onTap action is fired when it's children's onTap should be
**Describe the bug**
In Datagrid widget, I have an onTap on DataRow and a Collapsible, among other widgets as its children. When I click on Collapsible widget, so both actions gets fired i.e., Collapsible's and DataRow's.

**To Reproduce**
Use this example yaml to reproduce the issue:

```yaml
- DataGrid:
      DataColumns:
        - label: Number
        - label: Name
      item-template:
        data: [1, 2]
        name: dummyData
        template:
          DataRow:
            onTap:
              navigateScreen:
                name: forms
            children:
              - Text:
                  text: ${dummyData}
              - Collapsible:
                  expandIconPosition: "end"
                  items:
                    - key: "2"
                      label: Here is my label 2
                      children:
                        Text:
                          text: Here is my value 2
                  onCollapse:
                    executeCode: |
                      console.log('HelloC', data)
```

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Added blockage of unnecessary event passthrough on datarow on tap

## Issue ticket number and link

Closes #396 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

**Comment by sagardspeed2** [comment]
@evshi I tried many other ways to stop event prapogation, also tried by the adding parent div wrapper around the collapsible. but none of them works. except this one.

Need your idea or suggestion on this one

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx [hunk 1]
```diff
@@ -198,13 +198,26 @@ export const DataGrid: React.FC<GridProps> = (props) => {
...
+  };
+
   return (
     <div ref={containerRef}>
       <Table
         dataSource={namedData}
         key={resolvedWidgetId}
         onRow={(record, recordIndex) => {
-          return { onClick: () => onTapActionCallback(record, recordIndex) };
+          return { onClick: (e) => handleOnRowClick(e, record, recordIndex) };
```
Comment (evshi) [inline packages/runtime/src/widgets/DataGrid/DataGrid.tsx:220]: @sagardspeed2 I think you've misunderstood the issue. The issue is the event propagation from the Collapsible container. We need to wrap the Collapsible widget with a component and an event handler that stops event propagation.

**Review by evshi — CHANGES_REQUESTED** [review]

File: packages/runtime/src/widgets/DataGrid/DataGrid.tsx (hunk 1 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/DataGrid/DataGrid.tsx:220]: @evshi I did not understand the task in wrong way. I tried your suggested way in which we can wrap the collapsible inside an element and using that element, we need to stop the event propagation. But that thing is not working. When we add event stop propagation inside the wrapper, it also block the click event on the collapsible widget. so on the click on collapsible, it does not trigger the collapsible callback.

Comment (evshi) [reply packages/runtime/src/widgets/DataGrid/DataGrid.tsx]: @sagardspeed2 please double check what you're describing. It doesn't make sense that a container element would interfere with the event handling of its child element. I pushed a commit that resolves the issue with the approach I'm describing. Let me know how it differs from what you were seeing.

File: packages/runtime/src/widgets/Collapsible.tsx [hunk 2]
```diff
@@ -188,14 +188,21 @@ export const Collapsible: React.FC<CollapsibleProps> = (props) => {
...
-        expandIcon={expandIcon}
-        expandIconPosition={props.expandIconPosition}
-        items={collapsibleItems}
-        onChange={handleCollapsibleChange}
-      />
+      <div
+        onClick={(e) => e.stopPropagation()}
+        onKeyDown={(e) => e.stopPropagation()}
+        role="button"
+        tabIndex={0}
```
Comment (evshi) [inline packages/runtime/src/widgets/Collapsible.tsx:195]: These aren't necessary. If you're getting ESLint errors, let's ignore them for now.

**Review by evshi — APPROVED** [review]

**Pull request merged**